src/android/app/src/main/java/io/github/mandarine3ds/mandarine/ui/main/MainActivity.kt

```diff
@@ -171,9 +175,22 @@ class MainActivity : AppCompatActivity(), ThemeProvider {
 
     override fun onDestroy() {
         EmulationActivity.stopForegroundService(this)
+        NetPlayManager.shutdownNetwork()
```

Comment:
Doesn't it already shutdown on JNI_OnUnload?

Reply:
It doesn't hurt to have it as i was getting weid behavior without still running in the background 

This ensures that it can be properly closed down once the activity is done but you can test if it's really needed